src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java

```diff
@@ -95,32 +107,61 @@ private TypeDefinition getBaseEntity(XtextGraph xtextGraph, String entityTypenam
 
   private void validateBaseExtensionCompatibility(EntityMergingContext entityMergingContext) {
       TypeDefinition baseType = entityMergingContext.getBaseType();
-      TypeDefinition typeExtension = entityMergingContext.getTypeExtension();
-
-      // specification: directive @key(fields: _FieldSet!) repeatable on OBJECT | INTERFACE
-      if (!(isInterfaceTypeDefinition(baseType) && isInterfaceTypeDefinition(typeExtension)
-              || isObjectTypeDefinition(baseType) && isObjectTypeDefinition(typeExtension))) {
-          String errMsgTemplate =
-                  "Failed to merge entity extension to base type. typename%s, serviceNamespace=%s";
-          throw new TypeConflictException(
-                  format(
-                          errMsgTemplate,
-                          entityMergingContext.getTypename(),
-                          entityMergingContext.getServiceNamespace()));
-      }
+      TypeSystemDefinition typeSystemDefinition = entityMergingContext.getExtensionSystemDefinition();
+
+      if(typeSystemDefinition.getType() != null) {
+          TypeDefinition typeExtension = typeSystemDefinition.getType();
+
+          if (!(isInterfaceTypeDefinition(baseType) && isInterfaceTypeDefinition(typeExtension)
+                  || isObjectTypeDefinition(baseType) && isObjectTypeDefinition(typeExtension))) {
+              throw new TypeConflictException(
+                      format(
+                              TYPE_CONFLICT_ERR_MSG,
```

Comment:
I'm thinking of this message:
`Incompatible type definitions for Entity extension and Entity base type . typename%s, serviceNamespace=%s`